clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/job/CloneKubernetesJobAtomicOperation.groovy

```diff
@@ -0,0 +1,100 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.deploy.ops.job
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.deploy.DeploymentResult
+import com.netflix.spinnaker.clouddriver.kubernetes.api.KubernetesApiConverter
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.KubernetesUtil
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.description.job.CloneKubernetesJobAtomicOperationDescription
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.description.job.KubernetesJobRestartPolicy
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.exception.KubernetesResourceNotFoundException
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import io.fabric8.kubernetes.api.model.extensions.Job
+
+class CloneKubernetesJobAtomicOperation implements AtomicOperation<DeploymentResult> {
+  private static final String BASE_PHASE = "CLONE_JOB"
+
+  CloneKubernetesJobAtomicOperation(CloneKubernetesJobAtomicOperationDescription description) {
+    this.description = description
+  }
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  CloneKubernetesJobAtomicOperationDescription description
+
+  /*
+   * curl -X POST -H "Content-Type: application/json" -d  '[ { "cloneJob": { "source": { "jobName": "kub-test-v000" }, "account":  "my-kubernetes-account" } } ]' localhost:7002/kubernetes/ops
+  */
+  @Override
+  DeploymentResult operate(List priorOutputs) {
+    CloneKubernetesJobAtomicOperationDescription newDescription = cloneAndOverrideDescription()
+
+    task.updateStatus BASE_PHASE, "Initializing copy of job for " +
+      "${description.source.jobName}..."
+
+    RunKubernetesJobAtomicOperation deployer = new RunKubernetesJobAtomicOperation(newDescription)
+    DeploymentResult deploymentResult = deployer.operate(priorOutputs)
+
+    task.updateStatus BASE_PHASE, "Finished copying job for " +
+      "${description.source.jobName}."
+
+    task.updateStatus BASE_PHASE, "Finished copying job for " +
+      "${description.source.jobName}. " +
+      "New job = ${deploymentResult.deployedNames[0]}."
+
+    return deploymentResult
+  }
+
+  CloneKubernetesJobAtomicOperationDescription cloneAndOverrideDescription() {
+    CloneKubernetesJobAtomicOperationDescription newDescription = description.clone()
+
+    task.updateStatus BASE_PHASE, "Reading ancestor job ${description.source.jobName}..."
+
+    def credentials = description.credentials
+
+    description.source.namespace = description.source.namespace ?: "default"
```

Comment:
Feels wrong to mutate `description` here. Can you just operate on `newDescription` now (on L74-L75)?